protocol/0032-PRIM-price_monitoring.md

```diff
@@ -73,20 +77,26 @@ There are several reasons why this maximum is enforced.
     - the price protection auction commences and the transaction considered should be processed in this way (along with any other orders on the book and pending transactions that are valid for auction).
 
+- Per mark price update:
+  - when the next mark price update value is received it gets checked against the price monitoring engine,
+  - price monitoring engine sends back signal informing if the price protection auction should be triggered (and if so how long the auction period should be),
+  - if no trigger gets activated then the mark price update candidate gets accepted, included in the internal price history tracked by the price monitoring engine and mark-to-market process carries on as usual, otherwise:
+    - the price protection auction commences, any other mark price update candidates received before the auction ends get discarded, if the auction ends and results in a price than that price gets used as the next mark price, it gets included in the internal price history tracked by the price monitoring engine and the mark-to-market process carries on as usual.
+
 ### View from the price monitoring engine side
 
-Price monitoring engine will interface between the matching engine and the risk model. It will communicate with the matching engine every time a new transaction is processed (to check it its' arrival price should trigger an auction). It will communicate with the risk model with a predefined frequency to inform the risk model of the latest price history and obtain a new set of scaling factors used to calculate min/max prices from the reference price.
+Price monitoring engine will interface between the matching engine and the risk model. It will communicate with the matching engine every time a new transaction is processed (to check it its' arrival price should trigger an auction). It will communicate with the risk model with a predefined frequency to inform the risk model of the latest mark price history and obtain a new set of scaling factors used to calculate min/max prices from the reference price.
 
 Specifically:
 
-- Price monitoring engine averages (weighted by volume) all the prices received from the matching engine that have the same timestamp.
 - It periodically (in a predefined, deterministic way) sends the:
   - the probability level α,
   - period τ,
   - the associated reference price
 to the risk model and obtains the range of valid up/down price moves per each of the specified triggers. Please note that these can be expressed as either additive offsets or multiplicative factors depending on the risk model used. The reference price is the latest price such that it's at least τ old or the earliest available price should price history be shorter than τ.
-- It holds the history of volume weighted average prices looking back to the maximum τ configured in the market.
+- It holds the history of mark prices looking back to the maximum τ configured in the market.
```

Comment:
@davidsiska-vega you mentioned keeping VWAP for the last traded price, is that still something you want?

Reply:
I thought this would be contained within the mark price calculation logic, the price monitoring module would just keep track of the mark price (however it's calculated).